plotter/heat.go

```diff
@@ -0,0 +1,210 @@
+// Copyright ©2015 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package plotter
+
+import (
+	"image/color"
+	"math"
+
+	"github.com/gonum/plot"
+	"github.com/gonum/plot/palette"
+	"github.com/gonum/plot/vg"
+	"github.com/gonum/plot/vg/draw"
+)
+
+// GridXYZ describes three dimensional data where the X and Y
+// coordinates are arranged on a rectangular grid.
+type GridXYZ interface {
+	// Dims returns the dimensions of the grid.
+	Dims() (c, r int)
```

Comment:
FYI, I used Tarjan's algorithm to detect cycles in a dependency graph:
https://github.com/go-hep/fwk/blob/master/utils/tarjan/tarjan.go
http://en.wikipedia.org/wiki/Tarjan’s_strongly_connected_components_algorithm

hth,
-s